modules/login/php/passwordexpiry.class.inc

```diff
@@ -1,158 +0,0 @@
-        // the password.
-        //
-        // (This is also why the form element for the new password needs to
-        // be named "password" and why there's a hidden element named "login"
-        // in the smarty template.)
-        $this->tpl_data['username'] = $this->_username;
-        $this->form->addFormRule(array(&$this, '_validate'));
-    }
-
-
-    /**
-     * _validate validates the data that was submitted by the user. Validate
-     * for password-reset only checks that a username was submitted, not that
-     * it's valid, in order to prevent brute force phishing attacks.
-     *
-     * @param array $values The values that were submitted to the page.
-     *
-     * @return array (possibly empty) of errors
-     */
-    function _validate($values)
-    {
-        if (empty($values['password'])) {
-            // FIXME: The way error_message is displayed in the template is inherited
-            // from before this page was a "real" LORIS page, when it was all done
-            // from SinglePointLogin. As a result, the usual way of displaying errors
-            // by returning them in an array with the key of the element with the
-            // problem doesn't work.
-            $this->tpl_data['error_message'] = 'Cannot use an empty password';
-
-            // But we'll pretend it does anyways, so that validation still fails.
-            return array('password' => 'Cannot use an empty password');
-        }
-
-        $this->_user = \User::factory($this->_username);
-        $data        = $this->_user->getData();
-
-        if (password_verify($_POST['password'], $data['Password_hash'])) {
-            $this->tpl_data['error_message'] = 'You cannot keep the same password';
-            return array('password' => 'You cannot keep the same password');
-        }
-
-        if (!\User::isPasswordStrong(
-            $_POST['password'],
-            array(
-             $_POST['confirm'],
-             $data['UserID'],
-             $data['Email'],
-            ),
-            array(
-             '==',
-             '!=',
-             '!=',
-            )
-        )
-        ) {
-            $this->tpl_data['error_message'] = 'The password is weak, or'
-                . ' the passwords do not match';
-            return array('password' => 'Weak password.');
-        }
-        // Check if password is present in existing data breaches.
-        if (\User::pwnedPassword($_POST['password'])) {
-            $this->tpl_data['error_message'] = 'The password you chose is ' .
-                'too common. Please choose a more unique password.';
-            return array('password' => 'Password pwned.');
-        }
-
-        return true;
-    }
-
-    /**
-     * _process updates the password of the user and then redirects to the
-     * dashboard.
-     *
-     * @param array $values The values that were submitted to the page.
-     *
-     * @return void
-     */
-    function _process($values)
```

Comment:
surely we need this process function to update expire passwords? or did you move this somewhere else?

Reply:
@zaliqarosli that functionality was moved to the ajaxlogin.class.inc